ros/src/tl_detector/tl_detector.py

```diff
@@ -219,17 +220,37 @@ def get_next_tl_waypoint_index(self, stop_line_positions):
                 self.car_direction = -1
         else:
             self.car_direction = 1
+
         self.last_car_wp_idx = car_wp_idx
 
         # Find the closest upcoming stop line waypoint index in front of the car
         INDEX_DISTANCE_THRESHOLD = 150
+
+        light_dist = [0 < (self.car_direction *(line_index - car_wp_idx)) < INDEX_DISTANCE_THRESHOLD
+                        for line_index in self.stop_lines_wp_idxs]
+        
+        self.last_in_range = self.in_range
+
+        if any(light_dist):
+            tl_wp_idx = self.stop_lines_wp_idxs[np.argmax(light_dist)]
+            self.in_range = True
+            light = True
+        else:
+            self.in_range = False
+
+        if self.in_range and not self.last_in_range:
+            rospy.logwarn("tl_detector: TL in range SL_WP: {}, Car_WP: {}".format(tl_wp_idx, car_wp_idx))
+            self.last_in_range = self.in_range
+        
+        return tl_wp_idx
+        """
```

Comment:
i would prefer to remove the commeted code here, you can always check prev versions in git.